src/libraries/Microsoft.PowerFx.Core/Public/Config/UserInfo.cs

```diff
@@ -2,15 +2,142 @@
 // Licensed under the MIT license.
 
 using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Texl.Builtins;
+using Microsoft.PowerFx.Types;
 
 namespace Microsoft.PowerFx
 {
-    public class UserInfo : IUserInfo
+    /// <summary>
+    /// Definition for User object. 
+    /// - Names of C# members match the names we get in Power Fx's User object. 
+    /// - This has a curated set of user identifiers. This list can grow (hence this is a class, not an interface). 
+    /// - Host can specify which symbols are available, and leave these blank if not supported on this host. 
+    /// - Some information may require a network call to fetch, so we use task and virtual. Different properties may require different fetches, so let host override each property (rather than a single virtual).
+    /// </summary>
+    public class UserInfo
     {
-        public string FullName { get; set; }
+        /// <summary>
+        /// Name of the User object as it shows up in a Power Fx expression.
+        /// </summary>
+        public const string ObjectName = "User";
+
+        // Host should have specified the symbol wasn't available, so formulas shouldn't hit this at runtime. 
+        private static NotSupportedException Ex([CallerMemberName] string caller = null)
+        {
+            return new NotSupportedException($"User.{caller} is not supported.");
+        }
+
+        public virtual async Task<string> FullName(CancellationToken cancel = default) => throw Ex();
+
+        public virtual async Task<string> Email(CancellationToken cancel = default) => throw Ex();
+
+        /// <summary>
+        /// Dataverse User Table Id, coming from IExecutionContext.UserId. 
+        /// This is a key into 'systemuser' table.
+        /// </summary>
+        public virtual async Task<Guid> DataverseUserTableId(CancellationToken cancel = default) => throw Ex();
+
+        /// <summary>
+        /// TBD - For Cards / PVA. 
+        /// </summary>
+        public virtual async Task<Guid> BotMemberId(CancellationToken cancel = default) => throw Ex();
+
+        /// <summary>
+        /// Get the Power Fx type for a User object that has the given fields exposed. 
+        /// </summary>
+        /// <param name="fields">Valid subset of fields on user object.</param>
+        /// <returns></returns>
```

Comment:
should we write this function return user type in this comment instead of empty?